### ISSUE TITLE
Incorrectly detect arguments type with spread operator

### ISSUE THREAD

**Issue by wenerme** [Microsoft/TypeScript#18069]
**TypeScript Version:**  2.4.0 / nightly (2.5.0-dev.201xxxxx)

**Code**

Play [here](https://www.typescriptlang.org/play/#src=%0D%0Afunction%20out(a%3A%20number%2C%20b%3F%3A%20boolean)%3A%20%5Bnumber%2C%20boolean%5D%7B%0D%0A%20%20%20%20return%20%5B0%2C%20false%5D%3B%0D%0A%7D%0D%0A%0D%0Aout(...out(1))%3B%0D%0A)

```ts
function out(a: number, b?: boolean): [number, boolean]{
    return [0, false];
}

out(...out(1));
```

**Expected behavior:**

Correct syntax

**Actual behavior:**

Compile failed
Expected 1-2 arguments, but got a minimum of 0.

**Comment by ikatyang** [comment]
Seems duplicate of #4130.

**Comment by wenerme** [comment]
Close, in favor of https://github.com/Microsoft/TypeScript/issues/4130
